packages/coverage-istanbul/src/provider.ts

```diff
@@ -106,36 +106,47 @@ export class IstanbulCoverageProvider extends BaseCoverageProvider implements Co
     return { code, map }
   }
 
-  onAfterSuiteRun({ coverage }: AfterSuiteRunMeta) {
-    this.coverages.push(coverage)
+  /*
+   * Coverage and meta information passed from Vitest runners.
+   * Note that adding new entries here and requiring on those without
+   * backwards compatibility is a breaking change.
+   */
+  onAfterSuiteRun({ coverage, transformMode }: AfterSuiteRunMeta) {
+    if (transformMode !== 'web' && transformMode !== 'ssr')
+      throw new Error(`Invalid transform mode: ${transformMode}`)
+
+    this.coverages[transformMode].push(coverage as CoverageMapData)
   }
 
   async clean(clean = true) {
     if (clean && existsSync(this.options.reportsDirectory))
       await fs.rm(this.options.reportsDirectory, { recursive: true, force: true, maxRetries: 10 })
 
-    this.coverages = []
+    this.coverages = { ssr: [], web: [] }
   }
 
   async reportCoverage({ allTestsRun }: ReportContext = {}) {
-    const mergedCoverage: CoverageMap = this.coverages.reduce((coverage, previousCoverageMap) => {
-      const map = libCoverage.createCoverageMap(coverage)
-      map.merge(previousCoverageMap)
-      return map
-    }, libCoverage.createCoverageMap({}))
+    const coverageMaps = await Promise.all([
+      mergeAndTransformCoverage(this.coverages.ssr),
+      mergeAndTransformCoverage(this.coverages.web),
+    ])
+
+    if (this.options.all && allTestsRun) {
+      // Try to guess which transform mode should be used to uncovered files
+      const transformMode = this.coverages.ssr.length > this.coverages.web.length ? 'ssr' : 'web'
```

Comment:
Picking correct transform mode for uncovered files should not be that important. But is there a better way to do this? At this point we know the filenames, but they have not been run through Vite.

Reply:
We can import a matching environment and take it there I guess.

Reply:
How would that work in practice?

Looks like there's also a method on Vite-node for this: `project.vitenode.getTransformMode`. But even here we would first need to figure out which project would use this uncovered file. And if multiple ones do, there's no clear way to decide which one to use. 

I think it's OK to leave this as is.

Reply:
`groupFilesByEnv` looks promising. But still, we would need to figure out how to pick `project` for a file that has not been run. Anything that should be considered there? Simply pick the first `project` from `ctx.projects`?

Reply:
I think for those cases we should just use `Vites.runner`. We already document that untested files are loaded with a default config file: https://vitest.dev/guide/workspace.html#coverage

Reply:
`ctx.runner` seems to use `ctx.vitenode` underneath. 

https://github.com/vitest-dev/vitest/blob/23968580cecde0f8a4cd3caa54c894ca721fd033/packages/vitest/src/node/core.ts#L110-L116

Switched providers to use `ctx.vitenode.transformRequest` and `vitenode.fetchCache` for uncovered files now.